nens_auth_client/wso2.py

```diff
@@ -46,17 +49,47 @@ def logout_redirect(self, request, redirect_uri=None, login_after=False):
     def parse_access_token(self, token, claims_options=None, leeway=120):
         """Decode and validate a WSO2 access token and return its payload.
 
-        Note: this function just errors with the token claims in the error
-        message (so that we can figure out how we can actually validate the
-        token)
-
         Args:
           token (str): access token (base64 encoded JWT)
 
+        Returns:
+          claims (dict): the token payload
+
         Raises:
-          NotImplementedError: always
+          authlib.jose.errors.JoseError: if token is invalid
+          ValueError: if the key id is not present in the jwks.json
         """
-        raise NotImplementedError(decode_jwt(token))
+
+        # this is a copy from the _parse_id_token equivalent function
```

Comment:
Komt die `_parse_id_token()` uit onze code? Of uit authlib?
*Als* het uit authlib ofzo gekopieerd is zou ik even erbij zetten uit welke file het is gekopieerd.